
# Loading data







In this workshop we work through loading data. Once we have a curated and cleaned dataset we can work on generating insights from the data.

As a biologist you should be used to asking questions and gathering data. It is also important that you learn all aspects of the research process. This includes responsible data management (understanding data files & spreadsheet organisation, keeping data safe) and data analysis.

In this chapter we will look at the structure of data files, and how to read these with R. We will also continue to develop reproducible scripts. This means that we are writing scripts that are well organised and easy to read, and also making sure that our scripts are complete and capable of reproducing an analysis from start to finish. 

Transparency and reproducibility are key values in scientific research, when you analyse data in a reproducible way it means that others can understand and check your work. It also means that the most important person can benefit from your work, YOU! When you return to an analysis after even a short break, you will be thanking your earlier self if you have worked in a clear and reproducible way, as you can pick up right where you left off.  


## Meet the Penguins

We have chosen to continue working with a dataset you have been introduced to already - the Palmer Penguins dataset. Previously we loaded a cleaned dataset, very quickly using an R package. Today we will be working in a more realistic scenario - uploading our data from a spreadsheet into our R workspace.

This data, taken from the `palmerpenguins` (@R-palmerpenguins) package was originally published by @Antarctic. In our course we will work with real data that has been shared by other researchers.

The palmer penguins data contains size measurements, clutch observations, and blood isotope ratios for three penguin species observed on three islands in the Palmer Archipelago, Antarctica over a study period of three years.

<img src="images/gorman-penguins.jpg" title="Photo of three penguin species, Chinstrap, Gentoo, Adelie" alt="Photo of three penguin species, Chinstrap, Gentoo, Adelie" width="80%" style="display: block; margin: auto;" />

These data were collected from 2007 - 2009 by Dr. Kristen Gorman with the Palmer Station Long Term Ecological Research Program, part of the US Long Term Ecological Research Network. The data were imported directly from the Environmental Data Initiative (EDI) Data Portal, and are available for use by CC0 license (“No Rights Reserved”) in accordance with the Palmer Station Data Policy. We gratefully acknowledge Palmer Station LTER and the US LTER Network. Special thanks to Marty Downs (Director, LTER Network Office) for help regarding the data license & use. Here is our intrepid package co-author, Dr. Gorman, in action collecting some penguin data:

<img src="images/penguin-expedition.jpg" title="Photo of Dr Gorman in the middle of a flock of penguins" alt="Photo of Dr Gorman in the middle of a flock of penguins" width="80%" style="display: block; margin: auto;" />

Here is a map of the study site

<img src="images/antarctica-map.png" title="Antarctic Peninsula and the Palmer Field Station" alt="Antarctic Peninsula and the Palmer Field Station" width="80%" style="display: block; margin: auto;" />

## Activity 1: Organising our workspace

Before we can begin working with the data, we need to do some set-up. 

* Go to RStudio Cloud and open this week's R project

* Create the following folders using the + New Folder button in the Files tab

  * data
  * figures
  * scripts

<div class="warning">
<p>R is case-sensitive so type everything exactly as printed here</p>
</div>

Having these separate subfolders within our project helps keep things tidy, means it's harder to lose things, and lets you easily tell R exactly where to go to retrieve data.  

The next step of our workflow is to have a well organised project space. RStudio Cloud does a lot of the hard work for you, each new data project can be set up with its own Project space. 

We will define a project as a series of linked questions that uses one (or sometimes several) datasets. For example a coursework assignment for a particular module would be its own project, or eventually your final year research project. 

A Project will contain several files, possibly organised into sub-folders containing data, R scripts and final outputs. You might want to keep any information (wider reading) you have gathered that is relevant to your project.

<div class="figure" style="text-align: center">
<img src="images/project.png" alt="An example of a typical R project set-up" width="80%" />
<p class="caption">(\#fig:unnamed-chunk-6)An example of a typical R project set-up</p>
</div>

Within this project you will notice there is already one file *.Rproj*. This is an R project file, this is a very useful feature, it interacts with R to tell it you are working in a very specific place on the computer (in this case the cloud server we have dialed into). It means R will automatically treat the location of your project file as the 'working directory' and makes importing and exporting easier^[More on projects can be found in the R4DS book (https://r4ds.had.co.nz/workflow-projects.html)]. 

## Activity 2: Access our data

Now that we have a project workspace, we are ready to import some data.

* Use the link below to open a page in your browser with the data open

* Right-click Save As to download in csv format to your computer (Make a note of **where** the file is being downloaded to e.g. Downloads)

* Compare how the data looks in "raw" format to when you open the same data with Excel


```{=html}
<a href="https://raw.githubusercontent.com/UEABIO/data-sci-v1/main/book/files/penguins_raw.csv">
<button class="btn btn-success"><i class="fa fa-save"></i> Download penguin data as csv</button>
</a>
```


At first glance the data might look quite strange and messy. It has been stored as a **CSV** or comma-separated values file. CSV files are plain text files that can store large amounts of data, and can readily be imported into a spreadsheet or storage database. 

These files are the simplest form of database, no coloured cells, no formulae, no text formatting. Each row is a row of the data, each value of a row (previously separate columns) is separated by a comma. 

This file format helps us maintain an ethos **Keep Raw Data Raw** - 

In many cases, the captured or collected data may be unique and impossible to reproduce, such as measurements in a lab or field observations. For this reason, they should be protected from any possible loss. Every time a change is made to a raw data file it threatens the integrity of that information.

In practice, that means we only use our data file for data entry and storage. All the data manipulation, cleaning and analysis happens in R, using transparent and reproducible scripts.

<div class="info">
<p>We avoid saving files in the Excel format because they have a nasty habit of formatting or even losing data when the file gets large enough.</p>
<p>[<a href="https://www.theguardian.com/politics/2020/oct/05/how-excel-may-have-caused-loss-of-16000-covid-tests-in-england" class="uri">https://www.theguardian.com/politics/2020/oct/05/how-excel-may-have-caused-loss-of-16000-covid-tests-in-england</a>].</p>
<p>If you need to add data to a csv file, you can always open it in an Excel-like program and add more information, but remember to save it in the original csv format afterwards.</p>
</div>

<div class="figure" style="text-align: center">
<img src="images/excel_csv.png" alt="excel view, csv view" width="80%" />
<p class="caption">(\#fig:unnamed-chunk-9)Top image: Penguins data viewed in Excel, Bottom image: Penguins data in native csv format</p>
</div>

In raw format, each line of a CSV is separated by commas for different values. When you open this in a spreadsheet program like Excel it automatically converts those comma-separated values into tables and columns. 


## Activity 3: Upload our data

* The data is now in your Downloads folder on your computer

* We need to upload the data to our remote cloud-server (RStudio Cloud), select the upload files to server button in the Files tab

* Put your file into the data folder - if you make a mistake select the tickbox for your file, go to the cogs button and choose the option Move.

<div class="figure" style="text-align: center">
<img src="images/upload.png" alt="File tab" width="80%" />
<p class="caption">(\#fig:unnamed-chunk-10)Highlighted the buttons to upload files, and more options</p>
</div>

## Activity 4: Make a script

Let's now create a new R script file in which we will write instructions and store comments for manipulating data, developing tables and figures. Use the File > New Script menu item and select an R Script. 

Add the following:


```r
#___________________________----
# SET UP ---
## An analysis of the bill dimensions of male and female Adelie, Gentoo and Chinstrap penguins ----

### Data first published in  Gorman, KB, TD Williams, and WR Fraser. 2014. “Ecological Sexual Dimorphism and Environmental Variability Within a Community of Antarctic Penguins (Genus Pygoscelis).” PLos One 9 (3): e90081. https://doi.org/10.1371/journal.pone.0090081. ----
#__________________________----
```

Then load the following add-on package to the R script, just underneath these comments. Tidyverse isn't actually one package, but a bundle of many different packages that play well together - for example it *includes* `ggplot2` which we used in the last session, so we don't have to call that separately


```r
# PACKAGES ----
library(tidyverse) # tidy data packages
library(janitor) # cleans variable names
library(lubridate) # make sure dates are processed properly
#__________________________----
```

Save this file inside the scripts folder and call it `01_import_penguins_data.R`

### Document outline

## Activity 5: Read in data

Now we can read in the data. To do this we will use the function `read_csv()` that allows us to read in .csv files. There are also functions that allow you to read in .xlsx files and other formats, however in this course we will only use .csv files.

* First, we will create an object called `penguins_data` that contains the data in the `penguins_raw.csv` file. 


```r
# IMPORT DATA ----
penguins <- read_csv ("data/penguins_raw.csv")

head(penguins) # check the data has loaded, prints first 10 rows of dataframe
#__________________________----
```


<div class="danger">
<p>There is also a function called <code>read.csv()</code>. Be very careful NOT to use this function instead of <code>read_csv()</code> as they have different ways of naming columns.</p>
</div>

### Filepath

## Activity 6: Test yourself
### Head to Blackboard when complete and submit your answers to the quiz.

**Question 1.** In order to make your R project reproducible what filepath should you use? 

<select class='webex-select'><option value='blank'></option><option value=''>Absolute filepath</option><option value='answer'>Relative filepath</option></select>

**Question 2.** Which of these would be acceptable to include in a raw datafile? 

<select class='webex-select'><option value='blank'></option><option value=''>Highlighting some blocks of cells</option><option value=''>Excel formulae</option><option value='answer'>A column of observational notes from the field</option><option value=''>a mix of ddmmyy and yymmdd date formats</option></select>

**Question 3.** What should always be the first set of functions in our script? `?()`

<input class='webex-solveme nospaces' size='100' data-answer='["abind","acepack","actuar","ade4","afex","affy","affyio","AICcmodavg","airports","airway","alignfigR","amap","annotate","AnnotationDbi","AnnotationFilter","AnnotationForge","antiword","anytime","ape","apeglm","aplot","apyramid","arm","aroma.light","arrow","ash","AsioHeaders","askpass","assertthat","astsa","audio","babynames","backports","base64enc","base64url","batchtools","BayesFactor","bayesplot","bayestestR","bbmle","bdsmatrix","beachmat","beepr","beeswarm","BestAnimal","bestglm","BGLR","BH","bib2df","bibtex","bigmemory","bigmemory.sri","binom","Biobase","BiocFileCache","BiocGenerics","BiocManager","BiocParallel","BiocSingular","BiocStyle","BiocVersion","biomaRt","biomartr","biomformat","bios2mds","Biostatistics","Biostrings","biovizBase","bit","bit64","bitops","blme","blob","BMA","bmp","bookdown","brew","brglm2","bridgesampling","brio","Brobdingnag","broom","broom.helpers","broom.mixed","broomExtra","BSgenome","BSgenome.Hsapiens.UCSC.hg19","bslib","bsplus","BWStest","cachem","Cairo","calibrate","callr","car","carData","caret","Category","cellranger","checkmate","cherryblossom","chromote","chron","circular","ciTools","classInt","cli","clipr","cmprsk","coda","coefplot","coin","colorBlindness","colorblindr","colorfindr","colorspace","colourpicker","commonmark","conflicted","conj620labs","conquer","contfrac","corpcor","correlation","corrplot","covid19nytimes","covr","cowplot","cowsay","coxme","cpp11","crayon","credentials","crosstalk","crsmeta","crul","cubature","cubelyr","curl","cyclocomp","d3heatmap","dada2","data.table","DataExplorer","datapasta","datarobot","datasauRus","datawizard","DBI","dbplyr","DECIPHER","DelayedArray","DelayedMatrixStats","dendextend","DEoptimR","Deriv","desc","DescTools","DESeq","DESeq2","deSolve","details","devtools","DHARMa","DiagrammeR","dials","DiceDesign","dichromat","diffobj","digest","distill","distributional","DNAcopy","doParallel","doRNG","dotCall64","dotwhisker","downlit","downloader","downloadthis","dplyr","dqrng","drat","dsbox","DT","dtplyr","dtt","dunn.test","dygraphs","e1071","Ecdat","Ecfun","EDASeq","edgeR","effects","effectsize","effsize","ellipse","ellipsis","elliptic","emdbook","emmeans","EMT","english","enrichwith","EnsDb.Hsapiens.v86","ensembldb","epitools","equatiomatic","estimability","evaluate","Exact","exactRankTests","expint","expm","ExPosition","extrafont","extrafontdb","ez","ezpickr","factoextra","FactoMineR","fansi","farver","fastGHQuad","fastman","fastmap","fastqq","faux","fauxpas","fda","fds","feather","ff","fields","fission","fit.models","fitdistrplus","fivethirtyeight","flashClust","flexdashboard","flextable","FNN","fontawesome","forcats","foreach","forecast","forestplot","formatR","formattable","Formula","formula.tools","fortunes","fracdiff","fs","FSA","fst","furrr","futile.logger","futile.options","future","future.apply","gam","gap","gapminder","gargle","gbRd","gdsfmt","gdtools","GenABEL","GenABEL.data","gender","genefilter","geneplotter","generics","geneSLOPE","GenomeInfoDb","GenomeInfoDbData","genomes","GenomicAlignments","GenomicFeatures","GenomicRanges","GEOquery","geosphere","gert","ggalluvial","GGally","gganimate","ggbeeswarm","ggbio","ggbump","ggcats","ggcorrplot","ggdark","ggdist","ggeasy","ggeffects","ggExtra","ggfittext","ggforce","ggforestplot","ggfortify","gggenes","ggh4x","gghighlight","ggimage","ggmap","ggmsa","ggNestedBarChart","ggpattern","ggplot2","ggplotify","ggpol","ggpubr","ggraph","ggrepel","ggridges","ggsci","ggseqlogo","ggsignif","ggstance","ggstatsplot","ggtext","ggThemeAssist","ggthemes","ggtree","gh","ghclass","gifski","git2r","gitcreds","gld","glmmADMB","glmmTMB","glmnet","glmpca","globals","glossary","glue","gmp","GO.db","googledrive","googlesheets4","googleVis","GOstats","gower","GPfit","gradethis","graph","graphlayouts","greybox","gridBase","gridExtra","gridGeometry","gridGraphics","gridtext","groupedstats","grpreg","GSEABase","gt","gtable","gtExtras","gtools","gtsummary","Gviz","gvlma","GWASExactHW","GWASTools","Haplin","hardhat","haven","HDF5Array","HDInterval","hdrcde","heplots","here","hexbin","hexView","hglm","hglm.data","highlight","highr","Hmisc","hms","hrbrthemes","hscovar","htmlTable","htmltools","htmlwidgets","httpcode","httpuv","httr","humaniformat","hunspell","huxtable","hwriter","hypergeo","ids","igraph","infer","influenceR","ini","inline","InPosition","insight","ipmisc","ipred","IRanges","irlba","isoband","iterators","janeaustenr","janitor","jpeg","jquerylib","jsonlite","jtools","kableExtra","kernlab","kknn","klippy","km.ci","KMsurv","knitr","ks","kSamples","labeling","labelled","labelVector","lambda.r","lamW","LaplacesDemon","later","latexpdf","lattice","latticeExtra","lava","lavaan","lazyeval","lazyWeave","LCFdata","leaps","LearnBioconductor","learnr","lemon","lexicon","lhs","libcoin","lifecycle","limma","lintr","listenv","lm.beta","lme4","LMERConvenienceFunctions","lmerTest","lmom","lmtest","locfit","logistf","logspline","loo","lpSolve","lpSolveAPI","lsmeans","lubridate","magick","magrittr","manipulateWidget","maps","maptools","markdown","mathjaxr","matrixcalc","MatrixModels","matrixStats","maxstat","mboxr","mc2d","mclust","MCMCglmm","mcsm","memoise","memuse","merTools","metaBMA","metafor","metaplus","MetBrewer","mets","mgcv","MGDrivE","mgsub","mi","mice","microbenchmark","microbiome","mime","miniUI","minqa","misc3d","mitools","mixOmics","mlbench","mlmRev","mnormt","modelbased","modeldata","ModelMetrics","modelr","modeltools","motifRG","mplot","msa","multcomp","multcompView","multicool","multtest","MuMIn","munsell","mvtnorm","nadiv","networkD3","nloptr","NLP","NMF","nortest","numDeriv","oddsratio","officer","openintro","openNLP","openNLPdata","openssl","openxlsx","operator.tools","optimx","org.Hs.eg.db","OrganismDbi","osmdata","packrat","padr","pagedown","pairwiseAdonis","pairwiseComparisons","paletteer","palmerpenguins","pammtools","pander","parallelly","parameters","parsnip","partR2","patchwork","pbapply","pbivnorm","pbkrtest","pcaExplorer","pcaPP","PCAtools","pcr","pdftools","pec","performance","permute","PFAM.db","pheatmap","philentropy","phyloseq","piecewiseSEM","pillar","pinfsc50","pixmap","pkgbuild","pkgconfig","pkgload","pkgmaker","plogr","plot3D","plotly","plotrix","plotwidgets","pls","plyr","PMCMRplus","png","PoiClaClu","poissonreg","polspline","polyclip","polynom","poolr","pracma","praise","preprocessCore","prettydoc","prettyGraphs","prettyunits","prismatic","pROC","processx","prodlim","progress","PROJ","proj4","promises","ProtGenerics","proxy","pryr","ps","pscl","pspearman","psych","Publish","purrr","pwr","pzfx","qdap","qdapDictionaries","qdapRegex","qdapTools","qgam","qpdf","qqman","qqplotr","qtl","quadprog","quantmod","quantreg","quantsmooth","qvalue","R.cache","R.methodsS3","R.oo","R.utils","R2admb","R2jags","R2WinBUGS","R6","rafalib","rainbow","randomForest","ranger","rappdirs","rapportools","rARPACK","raster","rawr","RBGL","rbibutils","rcartocolor","rcmdcheck","RColorBrewer","rcompanion","Rcpp","RcppArmadillo","RcppEigen","RcppNumerical","RcppParallel","RcppProgress","RcppRoll","RCurl","Rdpack","reactable","reactablefmtr","reactR","readODS","readr","readxl","recipes","RefManageR","registry","relaimpo","rematch","rematch2","remotes","renv","RepeatABEL","report","ReportingTools","repr","reprex","reproj","reshape","reshape2","reticulate","rex","rgdal","rgl","Rglpk","RgoogleMaps","Rgraphviz","rhdf5","Rhdf5lib","Rhtslib","rio","riskRegression","rjags","rJava","rjson","rlang","rlist","RLRsim","rmarkdown","rmatio","rmdformats","Rmpfr","rms","rmsfact","rnaseqGene","rngtools","robmixglm","robotstxt","robust","robustbase","ROI","ROI.plugin.glpk","roll","rootSolve","rorcid","roxygen2","rphylopic","rprojroot","rrBLUP","rrcov","rsample","Rsamtools","rsconnect","RSpectra","rsq","RSQLite","rstan","rstanarm","rstantools","rstatix","rstudioapi","rsvd","rsvg","rtracklayer","Rtsne","Rttf2pt1","runner","RUVSeq","RVAideMemoire","rvcheck","rversions","rvest","S4Vectors","safeBinaryRegression","sandwich","sass","scales","scattermore","scatterplot3d","scholar","scico","sciplot","see","segmented","selectr","sem","SemiPar","sentimentr","seqinr","seqLogo","seqmagick","servr","sessioninfo","sf","sfsmisc","shades","shape","shiny","shinyAce","shinyBS","shinydashboard","shinyjs","shinystan","shinythemes","ShortRead","showtext","showtextdb","signal","simr","sitmo","sjlabelled","sjmisc","sjPlot","sjstats","skimr","slam","Sleuth3","slider","SLOPE","SMPracticals","snakecase","snow","SnowballC","SNPRelate","socviz","sommer","sourcetools","sp","spam","spaMM","SparseM","spiderbar","SQUAREM","StanHeaders","stargazer","statgenGWAS","statmod","statsExpressions","stringdist","stringi","stringr","striprtf","SummarizedExperiment","summarytools","SuppDists","survey","survminer","survMisc","sva","svglite","svgPanZoom","svMisc","sys","sysfonts","systemfonts","systemPipeR","syuzhet","TeachingDemos","tensorA","terra","testthat","texPreview","texreg","textclean","textreadr","textshape","TH.data","threejs","tibble","tictoc","tidyBF","tidygraph","tidymodels","tidyr","tidyselect","tidytext","tidytree","tidyverse","tiff","timeDate","timereg","timetk","tinytex","tippy","tis","tm","TMB","tmvnsim","tokenizers","topGO","transformr","treeio","treemap","triebeard","truncnorm","tseries","TTR","tufte","tune","tweenr","tximeta","tximport","tzdb","UEA5023Y","ukbabynames","units","unmarked","urca","urltools","usdata","useful","usethis","utf8","uuid","V8","VariantAnnotation","VCA","vcfR","vctrs","vegan","vembedr","VennDiagram","venneuler","VGAM","vip","vipor","viridis","viridisLite","visNetwork","visreg","vitae","vroom","vsn","waldo","warp","webexercises","webshot","websocket","wesanderson","whisker","winatr","withr","wordcloud","wordcloud2","workflows","writexl","WRS2","xaringan","xaringanBuilder","xaringanExtra","xaringanthemer","xfun","XML","xml2","xmlparsedata","xopen","xtable","xts","XVector","yaml","yardstick","yulab.utils","zeallot","zip","zlibbioc","zoo","base","boot","class","cluster","codetools","compiler","datasets","foreign","graphics","grDevices","grid","KernSmooth","lattice","MASS","Matrix","methods","mgcv","nlme","nnet","parallel","rpart","spatial","splines","stats","stats4","survival","tcltk","tools","translations","utils","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Users/phill/Documents/R/win-library/3.6","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","C:/Program Files/R/R-3.6.2/library","Combine Multidimensional Arrays","ACE and AVAS for Selecting Multiple Regression Transformations","Actuarial Functions and Heavy Tailed Distributions","Analysis of Ecological Data: Exploratory and Euclidean Methods\nin Environmental Sciences","Analysis of Factorial Experiments","Methods for Affymetrix Oligonucleotide Arrays","Tools for parsing Affymetrix data files","Model Selection and Multimodel Inference Based on (Q)AIC(c)","Data on Airports","RangedSummarizedExperiment for RNA-Seq in airway smooth muscle\ncells, by Himes et al PLoS One 2014","Visualizing Multiple Sequence Alignments with &apos;ggplot2&apos;","Another Multidimensional Analysis Package","Annotation for microarrays","Manipulation of SQLite-based annotations in Bioconductor","Facilities for Filtering Bioconductor Annotation Resources","Tools for building SQLite-based annotation data packages","Extract Text from Microsoft Word Documents","Anything to &apos;POSIXct&apos; or &apos;Date&apos; Converter","Analyses of Phylogenetics and Evolution","Approximate posterior estimation for GLM coefficients","Decorate a &apos;ggplot&apos; with Associated Information","Visualize Population Pyramids Aggregated by Age","Data Analysis Using Regression and Multilevel/Hierarchical\nModels","Light-Weight Methods for Normalization and Visualization of\nMicroarray Data using Only Basic R Data Types","Integration to &apos;Apache&apos; &apos;Arrow&apos;","David Scott&apos;s ASH Routines","&apos;Asio&apos; C++ Header Files","Safe Password Entry for R, Git, and SSH","Easy Pre and Post Assertions","Applied Statistical Time Series Analysis","Audio Interface for R","US Baby Names 1880-2017","Reimplementations of Functions Introduced Since R-3.0.0","Tools for base64 encoding","Fast and URL-Safe Base64 Encoder and Decoder","Tools for Computation on Batch Systems","Computation of Bayes Factors for Common Designs","Plotting for Bayesian Models","Understand and Describe Bayesian Models and Posterior\nDistributions","Tools for General Maximum Likelihood Estimation","Routines for Block Diagonal Symmetric Matrices","Compiling Bioconductor to Handle Each Matrix Type","Easily Play Notification Sounds on any Platform","The Bee Swarm Plot, an Alternative to Stripchart","What the Package Does (One Line, Title Case)","Best Subset GLM and Regression Utilities","Bayesian Generalized Linear Regression","Boost C++ Header Files","Parse a BibTeX File to a Data Frame","Bibtex Parser","Manage Massive Matrices with Shared Memory and Memory-Mapped\nFiles","A shared resource interface for Bigmemory Project packages","Binomial Confidence Intervals For Several Parameterizations","Biobase: Base functions for Bioconductor","Manage Files Across Sessions","S4 generic functions used in Bioconductor","Access the Bioconductor Project Package Repository","Bioconductor facilities for parallel evaluation","Singular Value Decomposition for Bioconductor Packages","Standard styles for vignettes and other Bioconductor documents","Set the appropriate version of Bioconductor packages","Interface to BioMart databases (i.e. Ensembl)","Genomic Data Retrieval","An interface package for the BIOM file format","From Biological Sequences to Multidimensional Scaling","Statistics Tutorials for Biologists","Efficient manipulation of biological strings","Basic graphic utilities for visualization of genomic data.","Classes and Methods for Fast Memory-Efficient Boolean Selections","A S3 Class for Vectors of 64bit Integers","Bitwise Operations","Bayesian Linear Mixed-Effects Models","A Simple S3 Class for Representing Vectors of Binary Data\n(&apos;BLOBS&apos;)","Bayesian Model Averaging","Read Windows Bitmap (BMP) Images","Authoring Books and Technical Documents with R Markdown","Templating Framework for Report Generation","Bias Reduction in Generalized Linear Models","Bridge Sampling for Marginal Likelihoods and Bayes Factors","Basic R Input Output","Very Large Numbers in R","Convert Statistical Objects into Tidy Tibbles","Helpers for Model Coefficients Tibbles","Tidying Methods for Mixed Models","Enhancements for &apos;broom&apos; and &apos;easystats&apos; Package Families","Software infrastructure for efficient representation of full\ngenomes and their SNPs","Full genome sequences for Homo sapiens (UCSC version hg19)","Custom &apos;Bootstrap&apos; &apos;Sass&apos; Themes for &apos;shiny&apos; and &apos;rmarkdown&apos;","Adds Functionality to the R Markdown + Shiny Bootstrap Framework","Baumgartner Weiss Schindler Test of Equal Distributions","Cache R Objects with Automatic Pruning","R Graphics Device using Cairo Graphics Library for Creating\nHigh-Quality Bitmap (PNG, JPEG, TIFF), Vector (PDF, SVG,\nPostScript) and Display (X11 and Win32) Output","Calibration of Scatterplot and Biplot Axes","Call R from R","Companion to Applied Regression","Companion to Applied Regression Data Sets","Classification and Regression Training","Category Analysis","Translate Spreadsheet Cell Ranges to Rows and Columns","Fast and Versatile Argument Checks","Cherry Blossom Run Race Results","Headless Chrome Web Browser Interface","Chronological Objects which can Handle Dates and Times","Circular Statistics","Confidence or Prediction Intervals, Quantiles, and Probabilities\nfor Statistical Models","Choose Univariate Class Intervals","Helpers for Developing Command Line Interfaces","Read and Write from the System Clipboard","Subdistribution Analysis of Competing Risks","Output Analysis and Diagnostics for MCMC","Plots Coefficients from Fitted Models","Conditional Inference Procedures in a Permutation Test Framework","Safe Color Set for Color Blindness","Simulate colorblindness in R figures","Extract Colors from Windows BMP, JPEG, PNG, TIFF, and SVG Format\nImages","A Toolbox for Manipulating and Assessing Colors and Palettes","A Colour Picker Tool for Shiny and for Selecting Colours in\nPlots","High Performance CommonMark and Github Markdown Rendering in R","An Alternative Conflict Resolution Strategy","What the Package Does (One Line, Title Case)","Convolution-Type Smoothed Quantile Regression","Continued Fractions","Efficient Estimation of Covariance and (Partial) Correlation","Methods for Correlation Analysis","Visualization of a Correlation Matrix","Pulls the Covid-19 Data from the New York Times Public Data\nSource","Test Coverage for Packages","Streamlined Plot Theme and Plot Annotations for &apos;ggplot2&apos;","Messages, Warnings, Strings with Ascii Animals","Mixed Effects Cox Models","A C++11 Interface for R&apos;s C Interface","Colored Terminal Output","Tools for Managing SSH and Git Credentials","Inter-Widget Interactivity for HTML Widgets","Extract Coordinate System Metadata","HTTP Client","Adaptive Multivariate Integration over Hypercubes","A Data Cube &apos;dplyr&apos; Backend","A Modern and Flexible Web Client for R","Cyclomatic Complexity of R Code","Interactive Heat Maps Using &apos;htmlwidgets&apos; and &apos;D3.js&apos;","Accurate, high-resolution sample inference from amplicon\nsequencing data","Extension of `data.frame`","Automate Data Exploration and Treatment","R Tools for Data Copy-Pasta","&apos;DataRobot&apos; Predictive Modeling API","Datasets from the Datasaurus Dozen","Easy Data Wrangling","R Database Interface","A &apos;dplyr&apos; Back End for Databases","Tools for curating, analyzing, and manipulating biological\nsequences","A unified framework for working transparently with on-disk and\nin-memory array-like datasets","Functions that Apply to Rows and Columns of &apos;DelayedMatrix&apos;\nObjects","Extending &apos;dendrogram&apos; Functionality in R","Differential Evolution Optimization in Pure R","Symbolic Differentiation","Manipulate DESCRIPTION Files","Tools for Descriptive Statistics","Differential gene expression analysis based on the negative\nbinomial distribution","Differential gene expression analysis based on the negative\nbinomial distribution","Solvers for Initial Value Problems of Differential Equations\n(&apos;ODE&apos;, &apos;DAE&apos;, &apos;DDE&apos;)","Create Details HTML Tag for Markdown and Package Documentation","Tools to Make Developing R Packages Easier","Residual Diagnostics for Hierarchical (Multi-Level / Mixed)\nRegression Models","Graph/Network Visualization","Tools for Creating Tuning Parameter Values","Designs of Computer Experiments","Color Schemes for Dichromats","Diffs for R Objects","Create Compact Hash Digests of R Objects","&apos;R Markdown&apos; Format for Scientific and Technical Writing","Vectorised Probability Distributions","DNA copy number data analysis","Foreach Parallel Adaptor for the &apos;parallel&apos; Package","Generic Reproducible Parallel Backend for &apos;foreach&apos; Loops","Enhanced Foreign Function Interface Supporting Long Vectors","Dot-and-Whisker Plots of Regression Results","Syntax Highlighting and Automatic Linking","Download Files over HTTP and HTTPS","Implement Download Buttons in &apos;rmarkdown&apos;","A Grammar of Data Manipulation","Fast Pseudo Random Number Generators","&apos;Drat&apos; R Archive Template","Data Science Course in a Box","A Wrapper of the JavaScript Library &apos;DataTables&apos;","Data Table Back-End for &apos;dplyr&apos;","Discrete Trigonometric Transforms","Dunn&apos;s Test of Multiple Comparisons Using Rank Sums","Interface to &apos;Dygraphs&apos; Interactive Time Series Charting Library","Misc Functions of the Department of Statistics, Probability\nTheory Group (Formerly: E1071), TU Wien","Data Sets for Econometrics","Functions for &apos;Ecdat&apos;","Exploratory Data Analysis and Normalization for RNA-Seq","Empirical Analysis of Digital Gene Expression Data in R","Effect Displays for Linear, Generalized Linear, and Other Models","Indices of Effect Size and Standardized Parameters","Efficient Effect Size Computation","Functions for Drawing Ellipses and Ellipse-Like Confidence\nRegions","Tools for Working with ...","Weierstrass and Jacobi Elliptic Functions","Support Functions and Data for \"Ecological Models and Data\"","Estimated Marginal Means, aka Least-Squares Means","Exact Multinomial Test: Goodness-of-Fit Test for Discrete\nMultivariate data","Translate Integers into English","Methods to Enrich R Objects with Extra Components","Ensembl based annotation package","Utilities to create and use Ensembl-based annotation databases","Epidemiology Tools","Transform Models into &apos;LaTeX&apos; Equations","Tools for Assessing Estimability of Linear Predictions","Parsing and Evaluation Tools that Provide More Details than the\nDefault","Unconditional Exact Test","Exact Distributions for Rank and Permutation Tests","Exponential Integral and Incomplete Gamma Function","Matrix Exponential, Log, &apos;etc&apos;","Exploratory Analysis with the Singular Value Decomposition","Tools for using fonts","Package for holding the database for the extrafont package","Easy Analysis and Visualization of Factorial Experiments","Easy Data Import Using GUI File Picker","Extract and Visualize the Results of Multivariate Data Analyses","Multivariate Exploratory Data Analysis and Data Mining","ANSI Control Sequence Aware String Functions","High Performance Colour Space Manipulation","Fast &apos;Rcpp&apos; Implementation of Gauss-Hermite Quadrature","A Fast Way to Create Manhattan Plots for GWAS Data","Fast Data Structures","Faster Generation of Quantile Quantile Plots with Large Samples","Simulation for Factorial Designs","HTTP Error Helpers","Functional Data Analysis","Functional Data Sets","R Bindings to the Feather &apos;API&apos;","Memory-Efficient Storage of Large Data on Disk and Fast Access\nFunctions","Tools for Spatial Data","RangedSummarizedExperiment for time course RNA-Seq of fission\nyeast in response to stress, by Leong et al., Nat Commun 2014.","Compare Fitted Models","Help to Fit of a Parametric Distribution to Non-Censored or\nCensored Data","Data and Code Behind the Stories and Interactives at\n&apos;FiveThirtyEight&apos;","Implementation of optimal hierarchical clustering","R Markdown Format for Flexible Dashboards","Functions for Tabular Reporting","Fast Nearest Neighbor Search Algorithms and Applications","Easily Work with &apos;Font Awesome&apos; Icons","Tools for Working with Categorical Variables (Factors)","Provides Foreach Looping Construct","Forecasting Functions for Time Series and Linear Models","Advanced Forest Plot Using &apos;grid&apos; Graphics","Format R Code Automatically","Create &apos;Formattable&apos; Data Structures","Extended Model Formulas","Programmatic Utilities for Manipulating Formulas, Expressions,\nCalls, Assignments and Other R Objects","R Fortunes","Fractionally Differenced ARIMA aka ARFIMA(P,d,q) Models","Cross-Platform File System Operations Based on &apos;libuv&apos;","Simple Fisheries Stock Assessment Methods","Lightning Fast Serialization of Data Frames","Apply Mapping Functions in Parallel using Futures","A Logging Utility for R","Futile Options Management","Unified Parallel and Distributed Processing in R for Everyone","Apply Function to Elements in Parallel using Futures","Generalized Additive Models","Genetic Analysis Package","Data from Gapminder","Utilities for Working with Google APIs","Utilities for processing Rd objects and files","R Interface to CoreArray Genomic Data Structure (GDS) Files","Utilities for Graphical Rendering","genome-wide SNP association analysis","Package contains data which is used by GenABEL example and test\nfunctions","Predict Gender from Names Using Historical Data","genefilter: methods for filtering genes from high-throughput\nexperiments","Graphics related functions for Bioconductor","Common S3 Generics not Provided by Base R Methods Related to\nModel Fitting","Genome-Wide Association Study with SLOPE","Utilities for manipulating chromosome names, including modifying\nthem to follow a particular naming style","Species and taxonomy ID look up tables used by GenomeInfoDb","Genome sequencing project metadata","Representation and manipulation of short genomic alignments","Conveniently import and query gene models","Representation and manipulation of genomic intervals","Get data from NCBI Gene Expression Omnibus (GEO)","Spherical Trigonometry","Simple Git Client for R","Alluvial Plots in &apos;ggplot2&apos;","Extension to &apos;ggplot2&apos;","A Grammar of Animated Graphics","Categorical Scatter (Violin Point) Plots","Visualization tools for genomic data","Bump Chart and Sigmoid Curves","The Geom you Always Wished for Adding Cats to ggplot2","Visualization of a Correlation Matrix using &apos;ggplot2&apos;","Dark Mode for &apos;ggplot2&apos; Themes","Visualizations of Distributions and Uncertainty","Easy Access to &apos;ggplot2&apos; Commands","Create Tidy Data Frames of Marginal Effects for &apos;ggplot&apos; from\nModel Outputs","Add Marginal Histograms to &apos;ggplot2&apos;, and More &apos;ggplot2&apos;\nEnhancements","Fit Text Inside a Box in &apos;ggplot2&apos;","Accelerating &apos;ggplot2&apos;","Forestplots of Measures of Effects and Their Confidence\nIntervals","Data Visualization Tools for Statistical Analysis Results","Draw Gene Arrow Maps in &apos;ggplot2&apos;","Hacks for &apos;ggplot2&apos;","Highlight Lines and Points in &apos;ggplot2&apos;","Use Image in &apos;ggplot2&apos;","Spatial Visualization with ggplot2","Plot Multiple Sequence Alignment using &apos;ggplot2&apos;","Create Nested, Grouped, Heirarchical Bar Charts with ggplot2","Geoms with Patterns","Create Elegant Data Visualisations Using the Grammar of Graphics","Convert Plot to &apos;grob&apos; or &apos;ggplot&apos; Object","Visualizing Social Science Data with &apos;ggplot2&apos;","&apos;ggplot2&apos; Based Publication Ready Plots","An Implementation of Grammar of Graphics for Graphs and Networks","Automatically Position Non-Overlapping Text Labels with\n&apos;ggplot2&apos;","Ridgeline Plots in &apos;ggplot2&apos;","Scientific Journal and Sci-Fi Themed Color Palettes for\n&apos;ggplot2&apos;","A &apos;ggplot2&apos; Extension for Drawing Publication-Ready Sequence\nLogos","Significance Brackets for &apos;ggplot2&apos;","Horizontal &apos;ggplot2&apos; Components","&apos;ggplot2&apos; Based Plots with Statistical Details","Improved Text Rendering Support for &apos;ggplot2&apos;","Add-in to Customize &apos;ggplot2&apos; Themes","Extra Themes, Scales and Geoms for &apos;ggplot2&apos;","an R package for visualization of tree and annotation data","&apos;GitHub&apos; &apos;API&apos;","Tools for Managing Classes on GitHub","Highest Quality GIF Encoder","Provides Access to Git Repositories","Query &apos;git&apos; Credentials from &apos;R&apos;","Estimation and Use of the Generalised (Tukey) Lambda\nDistribution","Generalized Linear Mixed Models using &apos;AD Model Builder&apos;","Generalized Linear Mixed Models using Template Model Builder","Lasso and Elastic-Net Regularized Generalized Linear Models","Dimension Reduction of Non-Normally Distributed Data","Identify Global Objects in R Expressions","Glossary of Statistical and Coding Terms","Interpreted String Literals","Multiple Precision Arithmetic","A set of annotation maps describing the entire Gene Ontology","An Interface to Google Drive","Access Google Sheets using the Sheets API V4","R Interface to Google Charts","Tools for manipulating GO and microarrays","Gower&apos;s Distance","Gaussian Processes Modeling","Tools for \"grading\" student exercises in learnr tutorials","graph: A package to handle graph data structures","Additional Layout Algorithms for Network Visualizations","Toolbox for Model Building and Forecasting","Integration of base and grid graphics","Miscellaneous Functions for \"Grid\" Graphics","Polygon Geometry in &apos;grid&apos;","Redraw Base Graphics Using &apos;grid&apos; Graphics","Improved Text Rendering Support for &apos;Grid&apos; Graphics","Grouped Statistical Analyses in a Tidy Way","Regularization Paths for Regression Models with Grouped\nCovariates","Gene set enrichment data structures and methods","Easily Create Presentation-Ready Display Tables","Arrange &apos;Grobs&apos; in Tables","A Collection of Helper Functions for the gt Package","Various R Programming Tools","Presentation-Ready Data Summary and Analytic Result Tables","Plotting data and annotation information along genomic\ncoordinates","Global Validation of Linear Models Assumptions","Exact Hardy-Weinburg testing for Genome Wide Association Studies","Tools for Genome Wide Association Studies","Analyzing Case-Parent Triad and/or Case-Control Data with SNP\nHaplotypes","Construct Modeling Packages","Import and Export &apos;SPSS&apos;, &apos;Stata&apos; and &apos;SAS&apos; Files","HDF5 backend for DelayedArray objects","Highest (Posterior) Density Intervals","Highest Density Regions and Conditional Density Estimation","Visualizing Hypothesis Tests in Multivariate Linear Models","A Simpler Way to Find Your Files","Hexagonal Binning Routines","Viewing Binary Files","Hierarchical Generalized Linear Models","Data for the &apos;hglm&apos; Package","Syntax Highlighter","Syntax Highlighting for R Source Code","Harrell Miscellaneous","Pretty Time of Day","Additional Themes, Theme Components and Utilities for &apos;ggplot2&apos;","Calculation of Covariance Between Markers for Half-Sib Families","Advanced Tables for Markdown/HTML","Tools for HTML","HTML Widgets for R","&apos;HTTP&apos; Status Code Helper","HTTP and WebSocket Server Library","Tools for Working with URLs and HTTP","A Parser for Human Names","High-Performance Stemmer, Tokenizer, and Spell Checker","Easily Create and Style Tables for LaTeX, HTML and Other Formats","HTML Writer - Outputs R objects in HTML format","The Gauss Hypergeometric Function","Generate Random Identifiers","Network Analysis and Visualization","Tidy Statistical Inference","Software Tools to Quantify Structural Importance of Nodes in a\nNetwork","Read and Write &apos;.ini&apos; Files","Functions to Inline C, C++, Fortran Function Calls from R","Inference Tests for ExPosition","Easy Access to Model Information for Various Model Objects","Miscellaneous Functions for Data Cleaning and Analysis","Improved Predictors","Foundation of integer range manipulation in Bioconductor","Fast Truncated Singular Value Decomposition and Principal\nComponents Analysis for Large Dense and Sparse Matrices","Generate Isolines and Isobands from Regularly Spaced Elevation\nGrids","Provides Iterator Construct","Jane Austen&apos;s Complete Novels","Simple Tools for Examining and Cleaning Dirty Data","Read and write JPEG images","Obtain &apos;jQuery&apos; as an HTML Dependency Object","A Simple and Robust JSON Parser and Generator for R","Analysis and Presentation of Social Scientific Data","Construct Complex Table with &apos;kable&apos; and Pipe Syntax","Kernel-Based Machine Learning Lab","Weighted k-Nearest Neighbors","Copy to Clipboard Buttons for R Markdown HTML Documents","Confidence intervals for the Kaplan-Meier estimator","Data sets from Klein and Moeschberger (1997), Survival Analysis","A General-Purpose Package for Dynamic Report Generation in R","Kernel Smoothing","K-Sample Rank Tests and their Combinations","Axis Labeling","Manipulating Labelled Data","Label Attributes for Atomic Vectors","Modeling Data with Functional Programming","Lambert-W Function","Complete Environment for Bayesian Inference","Utilities for Scheduling Functions to Execute Later with Event\nLoops","Convert Tables to PDF or PNG","Trellis Graphics for R","Extra Graphical Utilities Based on Lattice","Latent Variable Models","Latent Variable Analysis","Lazy (Non-Standard) Evaluation","LaTeX Wrappers for R Users","Data sets for package ``LMERConvenienceFunctions&apos;&apos;","Regression Subset Selection","Learning R / Bioconductor for Sequence Analysis","Interactive Tutorials for R","Freshing Up your &apos;ggplot2&apos; Plots","Lexicons for Text Analysis","Latin Hypercube Samples","Linear Test Statistics for Permutation Inference","Manage the Life Cycle of your Package Functions","Linear Models for Microarray Data","A &apos;Linter&apos; for R Code","Environments Behaving (Almost) as Lists","Add Standardized Regression Coefficients to lm-Objects","Linear Mixed-Effects Models using &apos;Eigen&apos; and S4","Model Selection and Post-Hoc Analysis for (G)LMER Models","Tests in Linear Mixed Effects Models","L-Moments","Testing Linear Regression Models","Local Regression, Likelihood and Density Estimation","Firth&apos;s Bias-Reduced Logistic Regression","Routines for Logspline Density Estimation","Efficient Leave-One-Out Cross-Validation and WAIC for Bayesian\nModels","Interface to &apos;Lp_solve&apos; v. 5.5 to Solve Linear/Integer Programs","R Interface to &apos;lp_solve&apos; Version 5.5.2.0","Least-Squares Means","Make Dealing with Dates a Little Easier","Advanced Graphics and Image-Processing in R","A Forward-Pipe Operator for R","Add Even More Interactivity to Interactive Charts","Draw Geographical Maps","Tools for Handling Spatial Objects","Render Markdown with the C Library &apos;Sundown&apos;","Using &apos;Mathjax&apos; in Rd Files","Collection of functions for matrix calculations","Modelling with Sparse And Dense Matrices","Functions that Apply to Rows and Columns of Matrices (and to\nVectors)","Maximally Selected Rank Statistics","Reading, Extracting, and Converting an Mbox File into a Tibble","Tools for Two-Dimensional Monte-Carlo Simulations","Gaussian Mixture Modelling for Model-Based Clustering,\nClassification, and Density Estimation","MCMC Generalised Linear Mixed Models","Functions for Monte Carlo Methods with R","Memoisation of Functions","Memory Estimation Utilities","Tools for Analyzing Mixed Effect Regression Models","Bayesian Model Averaging for Random and Fixed Effects\nMeta-Analysis","Meta-Analysis Package for R","Robust Meta-Analysis and Meta-Regression","Color Palettes Inspired by Works at the Metropolitan Museum of\nArt","Analysis of Multivariate Event Times","Mixed GAM Computation Vehicle with Automatic Smoothness\nEstimation","Mosquito Gene Drive Explorer","Safe, Multiple, Simultaneous String Substitution","Missing Data Imputation and Model Checking","Multivariate Imputation by Chained Equations","Accurate Timing Functions","Microbiome Analytics","Map Filenames to MIME Types","Shiny UI Widgets for Small Screens","Derivative-free optimization algorithms by quadratic\napproximation","Miscellaneous 3D Plots","Tools for Multiple Imputation of Missing Data","Omics Data Integration Project","Machine Learning Benchmark Problems","Examples from Multilevel Modelling Software Review","The Multivariate Normal and t Distributions, and Their Truncated\nVersions","Estimation of Model-Based Predictions, Contrasts and Means","Data Sets Used Useful for Modeling Packages","Rapid Calculation of Model Metrics","Modelling Functions that Work with the Pipe","Tools and Classes for Statistical Models","A package for discriminative motif discovery, designed for high\nthroughput sequencing dataset","Graphical Model Stability and Variable Selection Procedures","Multiple Sequence Alignment","Simultaneous Inference in General Parametric Models","Visualizations of Paired Comparisons","Permutations of Multisets in Cool-Lex Order","Resampling-based multiple hypothesis testing","Multi-Model Inference","Utilities for Using Munsell Colours","Multivariate Normal and t Distributions","(Non)Additive Genetic Relatedness Matrices","D3 JavaScript Network Graphs from R","R Interface to NLopt","Natural Language Processing Infrastructure","Algorithms and Framework for Nonnegative Matrix Factorization\n(NMF)","Tests for Normality","Accurate Numerical Derivatives","Odds Ratio Calculation for GAM(M)s & GLM(M)s","Manipulation of Microsoft Word and PowerPoint Documents","Data Sets and Supplemental Functions from &apos;OpenIntro&apos; Textbooks\nand Labs","Apache OpenNLP Tools Interface","Apache OpenNLP Jars and Basic English Language Models","Toolkit for Encryption, Signatures and Certificates Based on\nOpenSSL","Read, Write and Edit xlsx Files","Utilities for Working with R&apos;s Operators","Expanded Replacement and Extension of the &apos;optim&apos; Function","Genome wide annotation for Human","Software to enable the smooth interfacing of different database\npackages","Import &apos;OpenStreetMap&apos; Data as Simple Features or Spatial\nObjects","A Dependency Management System for Projects and their R Package\nDependencies","Quickly Get Datetime Data Ready for Analysis","Paginate the HTML Output of R Markdown with CSS for Print","Pairwise Multilevel Comparison using Adonis","Multiple Pairwise Comparison Tests","Comprehensive Collection of Color Palettes","Palmer Archipelago (Antarctica) Penguin Data","Piece-Wise Exponential Additive Mixed Modeling Tools for\nSurvival Analysis","An R &apos;Pandoc&apos; Writer","Enhancing the &apos;parallel&apos; Package","Processing of Model Parameters","A Common API to Modeling and Analysis Functions","Partitioning R2 in GLMMs","The Composer of Plots","Adding Progress Bar to &apos;*apply&apos; Functions","Vectorized Bivariate Normal CDF","Parametric Bootstrap, Kenward-Roger and Satterthwaite Based\nMethods for Test in Mixed Models","Interactive Visualization of RNA-seq Data Using a Principal\nComponents Approach","Robust PCA by Projection Pursuit","PCAtools: Everything Principal Components Analysis","Analyzing Real-Time Quantitative PCR Data","Text Extraction, Rendering and Converting of PDF Documents","Prediction Error Curves for Risk Prediction Models in Survival\nAnalysis","Assessment of Regression Models Performance","Functions for Generating Restricted Permutations of Data","A set of protein ID mappings for PFAM","Pretty Heatmaps","Similarity and Distance Quantification Between Probability\nFunctions","Handling and analysis of high-throughput microbiome census data","Piecewise Structural Equation Modeling","Coloured Formatting for Columns","Sequence (&apos;FASTA&apos;), Annotation (&apos;GFF&apos;) and Variants (&apos;VCF&apos;) for\n17 Samples of &apos;P. Infestans\" and 1 &apos;P. Mirabilis&apos;","Bitmap Images (``Pixel Maps&apos;&apos;)","Find Tools Needed to Build R Packages","Private Configuration for &apos;R&apos; Packages","Simulate Package Installation and Attach","Development Utilities for R Packages","The &apos;plog&apos; C++ Logging Library","Plotting Multi-Dimensional Data","Create Interactive Web Graphics via &apos;plotly.js&apos;","Various Plotting Functions","Spider Plots, ROC Curves, Pie Charts and More for Use in Other\nPlots","Partial Least Squares and Principal Component Regression","Tools for Splitting, Applying and Combining Data","Calculate Pairwise Multiple Comparisons of Mean Rank Sums\nExtended","Read and write PNG images","Classification and Clustering of Sequencing Data Based on a\nPoisson Model","Model Wrappers for Poisson Regression","Polynomial Spline Routines","Polygon Clipping","A Collection of Functions to Implement a Class for Univariate\nPolynomial Manipulations","Methods for Pooling P-Values from (Dependent) Tests","Practical Numerical Math Functions","Praise Users","A collection of pre-processing functions","Creating Pretty Documents from R Markdown","Publication-Quality Graphics","Pretty, Human Readable Formatting of Quantities","Color Manipulation Tools","Display and Analyze ROC Curves","Execute and Control System Processes","Product-Limit Estimation for Censored Event History Analysis","Terminal Progress Bars","Generic Coordinate System Transformations Using &apos;PROJ&apos;","A simple interface to the PROJ.4 cartographic projections\nlibrary","Abstractions for Promise-Based Asynchronous Programming","S4 generic functions for Bioconductor proteomics infrastructure","Distance and Similarity Measures","Tools for Computing on the Language","List, Query, Manipulate System Processes","Political Science Computational Laboratory","Spearman&apos;s rank correlation test","Procedures for Psychological, Psychometric, and Personality\nResearch","Format Output of Various Routines in a Suitable Way for Reports\nand Publication","Functional Programming Tools","Basic Functions for Power Analysis","Read and Write &apos;GraphPad Prism&apos; Files","Bridging the Gap Between Qualitative Data and Quantitative\nAnalysis","Dictionaries and Word Lists for the &apos;qdap&apos; Package","Regular Expression Removal, Extraction, and Replacement Tools","Tools for the &apos;qdap&apos; Package","Smooth Additive Quantile Regression Models","Split, Combine and Compress PDF Files","Q-Q and Manhattan Plots for GWAS Data","Quantile-Quantile Plot Extensions for &apos;ggplot2&apos;","Tools for Analyzing QTL Experiments","Functions to Solve Quadratic Programming Problems","Quantitative Financial Modelling Framework","Quantile Regression","Quantile smoothing and genomic visualization of array data","Q-value estimation for false discovery rate control","Fast and Light-Weight Caching (Memoization) of Objects and\nResults to Speed Up Computations","S3 Methods Simplified","R Object-Oriented Programming with or without References","Various Programming Utilities","&apos;ADMB&apos; to R Interface Functions","Using R to Run &apos;JAGS&apos;","Running &apos;WinBUGS&apos; and &apos;OpenBUGS&apos; from &apos;R&apos; / &apos;S-PLUS&apos;","Encapsulated Classes with Reference Semantics","Convenience Functions for Routine Data Exploration","Bagplots, Boxplots and Rainbow Plots for Functional Data","Breiman and Cutler&apos;s Random Forests for Classification and\nRegression","A Fast Implementation of Random Forests","Application Directories: Determine Where to Save Data, Caches,\nand Logs","Miscellaneous (stats) helper functions with sane defaults for\nreporting","Solvers for Large Scale Eigenvalue and SVD Problems","Geographic Data Analysis and Modeling","rawr miscellaneous","An interface to the BOOST graph library","Convert Between Bibliography Formats","&apos;CARTOColors&apos; Palettes","Run &apos;R CMD check&apos; from &apos;R&apos; and Capture Results","ColorBrewer Palettes","Functions to Support Extension Education Program Evaluation","Seamless R and C++ Integration","&apos;Rcpp&apos; Integration for the &apos;Armadillo&apos; Templated Linear Algebra\nLibrary","&apos;Rcpp&apos; Integration for the &apos;Eigen&apos; Templated Linear Algebra\nLibrary","&apos;Rcpp&apos; Integration for Numerical Computing Libraries","Parallel Programming Tools for &apos;Rcpp&apos;","An Interruptible Progress Bar with OpenMP Support for C++ in R\nPackages","Efficient Rolling / Windowed Operations","General Network (HTTP/FTP/...) Client Interface for R","Update and Manipulate Rd Documentation Objects","Interactive Data Tables Based on &apos;React Table&apos;","Streamlined Table Styling and Formatting for Reactable","React Helpers","Read and Write ODS Files","Read Rectangular Text Data","Read Excel Files","Preprocessing Tools to Create Design Matrices","Straightforward &apos;BibTeX&apos; and &apos;BibLaTeX&apos; Bibliography Management","Infrastructure for R Package Registries","Relative Importance of Regressors in Linear Models","Match Regular Expressions with a Nicer &apos;API&apos;","Tidy Output from Regular Expression Matching","R Package Installation from Remote Repositories, Including\n&apos;GitHub&apos;","Project Environments","GWAS for Multiple Observations on Related Individuals","Automated Reporting of Results and Statistical Models","Tools for making reports in various formats","Serializable Representations","Prepare Reproducible Example Code via the Clipboard","Coordinate System Transformations for Generic Map Data","Flexibly Reshape Data","Flexibly Reshape Data: A Reboot of the Reshape Package","Interface to &apos;Python&apos;","Friendly Regular Expressions","Bindings for the &apos;Geospatial&apos; Data Abstraction Library","3D Visualization Using OpenGL","R/GNU Linear Programming Kit Interface","Overlays on Static Maps","Provides plotting capabilities for R graph objects","R Interface to HDF5","hdf5 library as an R package","HTSlib high-throughput sequencing library as an R package","A Swiss-Army Knife for Data I/O","Risk Regression Models and Prediction Scores for Survival\nAnalysis with Competing Risks","Bayesian Graphical Models using MCMC","Low-Level R to Java Interface","JSON for R","Functions for Base Types and Core R and &apos;Tidyverse&apos; Features","A Toolbox for Non-Tabular Data Manipulation","Exact (Restricted) Likelihood Ratio Tests for Mixed and Additive\nModels","Dynamic Documents for R","Read and Write &apos;Matlab&apos; Files","HTML Output Formats and Templates for &apos;rmarkdown&apos; Documents","R MPFR - Multiple Precision Floating-Point Reliable","Regression Modeling Strategies","Amazing Random Facts About the World&apos;s Greatest Hacker","RNA-seq workflow: gene-level exploratory analysis and\ndifferential expression","Utility Functions for Working with Random Number Generators","Robust Generalized Linear Models (GLM) using Mixtures","A &apos;robots.txt&apos; Parser and &apos;Webbot&apos;/&apos;Spider&apos;/&apos;Crawler&apos;\nPermissions Checker","Port of the S+ \"Robust Library\"","Basic Robust Statistics","R Optimization Infrastructure","&apos;ROI&apos; Plug-in &apos;GLPK&apos;","Rolling and Expanding Statistics","Nonlinear Root Finding, Equilibrium and Steady-State Analysis of\nOrdinary Differential Equations","Interface to the &apos;Orcid.org&apos; API","In-Line Documentation for R","Get &apos;Silhouettes&apos; of &apos;Organisms&apos; from &apos;Phylopic&apos;","Finding Files in Project Subdirectories","Ridge Regression and Other Kernels for Genomic Selection","Scalable Robust Estimators with High Breakdown Point","General Resampling Infrastructure","Binary alignment (BAM), FASTA, variant call (BCF), and tabix\nfile import","Deployment Interface for R Markdown Documents and Shiny\nApplications","Solvers for Large-Scale Eigenvalue and SVD Problems","R-Squared and Related Measures","&apos;SQLite&apos; Interface for R","R Interface to Stan","Bayesian Applied Regression Modeling via Stan","Tools for Developing R Packages Interfacing with &apos;Stan&apos;","Pipe-Friendly Framework for Basic Statistical Tests","Safely Access the RStudio API","Randomized Singular Value Decomposition","Render SVG Images into PDF, PNG, PostScript, or Bitmap Arrays","R interface to genome annotation files and the UCSC genome\nbrowser","T-Distributed Stochastic Neighbor Embedding using a Barnes-Hut\nImplementation","&apos;ttf2pt1&apos; Program","Running Operations for Vectors","Remove Unwanted Variation from RNA-Seq Data","Testing and Plotting Procedures for Biostatistics","R/Package Version Check","Query &apos;R&apos; Versions, Including &apos;r-release&apos; and &apos;r-oldrel&apos;","Easily Harvest (Scrape) Web Pages","Foundation of vector-like and list-like containers in\nBioconductor","Safe Binary Regression","Robust Covariance Matrix Estimators","Syntactically Awesome Style Sheets (&apos;Sass&apos;)","Scale Functions for Visualization","Scatterplots with More Points","3D Scatter Plot","Analyse Citation Data from Google Scholar","Colour Palettes Based on the Scientific Colour-Maps","Scientific Graphing Functions for Factorial Designs","Visualisation Toolbox for &apos;easystats&apos; and Extra Geoms, Themes\nand Color Palettes for &apos;ggplot2&apos;","Regression Models with Break-Points / Change-Points Estimation","Translate CSS Selectors to XPath Expressions","Structural Equation Models","Semiparametic Regression","Calculate Text Polarity Sentiment","Biological Sequences Retrieval and Analysis","Sequence logos for DNA sequence alignments","Sequence Manipulation Utilities","A Simple HTTP Server to Serve Static Files or Dynamic Documents","R Session Information","Simple Features for R","Utilities from &apos;Seminar fuer Statistik&apos; ETH Zurich","Simple Colour Manipulation","Functions for Plotting Graphical Shapes, Colors","Web Application Framework for R","Ace Editor Bindings for Shiny","Twitter Bootstrap Components for Shiny","Create Dashboards with &apos;Shiny&apos;","Easily Improve the User Experience of Your Shiny Apps in Seconds","Interactive Visual and Numerical Diagnostics and Posterior\nAnalysis for Bayesian Models","Themes for Shiny","FASTQ input and manipulation","Using Fonts More Easily in R Graphs","Font Files for the &apos;showtext&apos; Package","Signal Processing","Power Analysis for Generalised Linear Mixed Models by Simulation","Parallel Pseudo Random Number Generator (PPRNG) &apos;sitmo&apos; Header\nFiles","Labelled Data Utility Functions","Data and Variable Transformation Functions","Data Visualization for Statistics in Social Science","Collection of Convenient Functions for Common Statistical\nComputations","Compact and Flexible Summaries of Data","Sparse Lightweight Arrays and Matrices","Data Sets from Ramsey and Schafer&apos;s \"Statistical Sleuth (3rd\nEd)\"","Sliding Window Functions","Sorted L1 Penalized Estimation","Practicals for Use with Davison (2003) Statistical Models","Convert Strings into any Case","Simple Network of Workstations","Snowball Stemmers Based on the C &apos;libstemmer&apos; UTF-8 Library","Parallel Computing Toolset for Relatedness and Principal\nComponent Analysis of SNP Data","Utility Functions and Data Sets for Data Visualization","Solving Mixed Model Equations in R","Tools for Reading, Tokenizing and Parsing R Code","Classes and Methods for Spatial Data","SPArse Matrix","Mixed-Effect Models, Particularly Spatial Models","Sparse Linear Algebra","Parse and Test Robots Exclusion Protocol Files and Rules","Squared Extrapolation Methods for Accelerating EM-Like Monotone\nAlgorithms","C++ Header Files for Stan","Well-Formatted Regression and Summary Statistics Tables","Genome Wide Association Studies","Statistical Modeling","Expressions and Dataframes with Statistical Details","Approximate String Matching, Fuzzy Text Search, and String\nDistance Functions","Character String Processing Facilities","Simple, Consistent Wrappers for Common String Operations","Extract Text from RTF File","SummarizedExperiment container","Tools to Quickly and Neatly Summarize Data","Supplementary Distributions","Analysis of Complex Survey Samples","Drawing Survival Curves using &apos;ggplot2&apos;","Miscellaneous Functions for Survival Data","Surrogate Variable Analysis","An &apos;SVG&apos; Graphics Device","R &apos;Htmlwidget&apos; to Add Pan and Zoom to Almost any R Graphic","&apos;SciViews&apos; - Miscellaneous Functions","Powerful and Reliable Tools for Running System Commands in R","Loading Fonts into R","System Native Font Finding","systemPipeR: NGS workflow and report generation environment","Extracts Sentiment and Sentiment-Derived Plot Arcs from Text","Demonstrations for Teaching and Learning","Advanced Tensor Arithmetic with Named Indices","Spatial Data Analysis","Unit Testing for R","Compile and Preview Snippets of &apos;LaTeX&apos;","Conversion of R Regression Output to LaTeX or HTML Tables","Text Cleaning Tools","Read Text Documents into R","Tools for Reshaping Text","TH&apos;s Data Archive","Interactive 3D Scatter Plots, Networks and Globes","Simple Data Frames","Functions for timing R scripts, as well as implementations of\nStack and List structures.","Tidy Wrapper for &apos;BayesFactor&apos; Package","A Tidy API for Graph Manipulation","Easily Install and Load the &apos;Tidymodels&apos; Packages","Tidy Messy Data","Select from a Set of Strings","Text Mining using &apos;dplyr&apos;, &apos;ggplot2&apos;, and Other Tidy Tools","A Tidy Tool for Phylogenetic Tree Data Manipulation","Easily Install and Load the &apos;Tidyverse&apos;","Read and Write TIFF Images","Rmetrics - Chronological and Calendar Objects","Flexible Regression Models for Survival Data","A Tool Kit for Working with Time Series in R","Helper Functions to Install and Maintain TeX Live, and Compile\nLaTeX Documents","Add Tooltips to &apos;R markdown&apos; Documents or &apos;Shiny&apos; Apps","Time Indexes and Time Indexed Series","Text Mining Package","Template Model Builder: A General Random Effect Tool Inspired by\n&apos;ADMB&apos;","Truncated Multivariate Normal Simulation","Fast, Consistent Tokenization of Natural Language Text","Enrichment Analysis for Gene Ontology","Polygon and Path Transformations","Base Classes and Functions for Phylogenetic Tree Input and\nOutput","Treemap Visualization","&apos;Radix&apos; Trees in &apos;Rcpp&apos;","Truncated Normal Distribution","Time Series Analysis and Computational Finance","Technical Trading Rules","Tufte&apos;s Styles for R Markdown Documents","Tidy Tuning Tools","Interpolate Data for Smooth Animations","Transcript Quantification Import with Automatic Metadata","Import and summarize transcript-level estimates for transcript-\nand gene-level analysis","Time Zone Database Information","What the Package Does (Title Case)","UK Baby Names Data","Measurement Units for R Vectors","Models for Data from Unmarked Animals","Unit Root and Cointegration Tests for Time Series Data","Vectorised Tools for URL Handling and Parsing","Data on the States and Counties of the United States","A Collection of Handy, Useful Functions","Automate Package and Project Setup","Unicode Text Processing","Tools for Generating and Handling of UUIDs","Embedded JavaScript and WebAssembly Engine for R","Annotation of Genetic Variants","Variance Component Analysis","Manipulate and Visualize VCF Data","Vector Helpers","Community Ecology Package","Embed Video in HTML","Generate High-Resolution Venn and Euler Plots","Venn and Euler Diagrams","Vector Generalized Linear and Additive Models","Variable Importance Plots","Plot Categorical Data Using Quasirandom Noise and Density\nEstimates","Colorblind-Friendly Color Maps for R","Colorblind-Friendly Color Maps (Lite Version)","Network Visualization using &apos;vis.js&apos; Library","Visualization of Regression Models","Curriculum Vitae for R Markdown","Read and Write Rectangular Text Data Quickly","Variance stabilization and calibration for microarray data","Find Differences Between R Objects","Group Dates","Create Interactive Web Exercises in &apos;R Markdown&apos; (Formerly\n&apos;webex&apos;)","Take Screenshots of Web Pages","&apos;WebSocket&apos; Client Library","A Wes Anderson Palette Generator","{{mustache}} for R, Logicless Templating","What the Package Does (Title Case)","Run Code &apos;With&apos; Temporarily Modified Global State","Word Clouds","Create Word Cloud by &apos;htmlwidget&apos;","Modeling Workflows","Export Data Frames to Excel &apos;xlsx&apos; Format","A Collection of Robust Statistical Methods","Presentation Ninja","Build &apos;xaringan&apos; Slides To Many Different Outputs","Extras and Extensions for &apos;xaringan&apos; Slides","Custom &apos;xaringan&apos; CSS Themes","Miscellaneous Functions to Support Packages Maintained by &apos;Yihui\nXie&apos;","Tools for Parsing and Generating XML Within R and S-Plus","Parse XML","Parse Data of &apos;R&apos; Code as an &apos;XML&apos; Tree","Open System Files, &apos;URLs&apos;, Anything","Export Tables to LaTeX or HTML","eXtensible Time Series","Foundation of external vector representation and manipulation in\nBioconductor","Methods to Convert R Data to YAML and Back","Tidy Characterizations of Model Performance","Supporting Functions for Packages Maintained by &apos;YuLab-SMU&apos;","Multiple, Unpacking, and Destructuring Assignment","Cross-Platform &apos;zip&apos; Compression","An R packaged zlib-1.2.5","S3 Infrastructure for Regular and Irregular Time Series (Z&apos;s\nOrdered Observations)","The R Base Package","Bootstrap Functions (Originally by Angelo Canty for S)","Functions for Classification","\"Finding Groups in Data\": Cluster Analysis Extended Rousseeuw et\nal.","Code Analysis Tools for R","The R Compiler Package","The R Datasets Package","Read Data Stored by &apos;Minitab&apos;, &apos;S&apos;, &apos;SAS&apos;, &apos;SPSS&apos;, &apos;Stata&apos;,\n&apos;Systat&apos;, &apos;Weka&apos;, &apos;dBase&apos;, ...","The R Graphics Package","The R Graphics Devices and Support for Colours and Fonts","The Grid Graphics Package","Functions for Kernel Smoothing Supporting Wand & Jones (1995)","Trellis Graphics for R","Support Functions and Datasets for Venables and Ripley&apos;s MASS","Sparse and Dense Matrix Classes and Methods","Formal Methods and Classes","Mixed GAM Computation Vehicle with Automatic Smoothness\nEstimation","Linear and Nonlinear Mixed Effects Models","Feed-Forward Neural Networks and Multinomial Log-Linear Models","Support for Parallel computation in R","Recursive Partitioning and Regression Trees","Functions for Kriging and Point Pattern Analysis","Regression Spline Functions and Classes","The R Stats Package","Statistical Functions using S4 Classes","Survival Analysis","Tcl/Tk Interface","Tools for Package Development","The R Translations Package","The R Utils Package"]'/>

**Question 4.** When reading in data to R we should use

<select class='webex-select'><option value='blank'></option><option value='answer'>read_csv()</option><option value=''>read.csv()</option></select>



